lib/trip_dispatcher.rb

```diff
@@ -27,14 +35,51 @@ def inspect
               #{passengers.count} passengers>"
     end
 
+    def find_available_drivers
+      available_drivers = []
+      @drivers.each do |driver|
+        if driver.status == :AVAILABLE
+          available_drivers << driver
+        end
+      end
+      return available_drivers
+    end
+
+    # Wave 3: creates a trip and assigns an available driver
+    def request_trip(passenger_id)
+      dispatch_driver = find_available_drivers
+      if dispatch_driver.length == 0
+        raise ArgumentError.new("No available drivers")
+      end
+      trip = Trip.new(
+        id: @trips.last.id + 1,
+        passenger: find_passenger(passenger_id),
+        passenger_id: passenger_id,
+        start_time: Time.now,
+        end_time: nil,
+        cost: nil,
+        rating: nil,
+        driver: dispatch_driver[0],
+        driver_id: dispatch_driver[0].id,
+      )
+      dispatch_driver[0].add_trip(trip)
+      dispatch_driver[0].status = :UNAVAILABLE
+      trip.passenger.add_trip(trip)
+      @trips << trip
+      trip.connect(trip.passenger)
+      trip.connect_driver(dispatch_driver[0])
```

Comment:
One last thing to think about:

A line that sticks out of this logic is setting the driver's status to unavailable. But this should ALWAYS happen every time we connect a trip and a driver. Can the logic for changing a driver's status to unavilable live in the same spot as when a trip and driver get connected, or at least get moved out of this `TripDispatcher` method?